sphinx/directives/code.py

```diff
@@ -219,6 +219,9 @@ def run(self):
 
         lines = self.read_with_encoding(filename, document,
                                         codec_info, encoding)
+        if not lines:
+            lines = ['']
```

Comment:
`Directive.run()` method should return list of *node*.
http://docutils.sourceforge.net/docs/howto/rst-directives.html#the-directive-class